### PR TITLE
feat: Adding SCM option for init command

### PR DESCRIPTION
**Pull request by alvgaona**
I'm sorry for my boldness. I sometimes work with multiple SCMs and found that this command lacked this feature. Let me know if it's aligned with the project. Feel free to close this PR, otherwise. 😄

### CODE REVIEW

File: src/cli/init.rs [hunk 1]
```diff
@@ -53,6 +53,10 @@ pub struct Args {
     // BREAK (0.27.0): Remove this option from the cli in favor of the `format` option.
     #[arg(long, conflicts_with_all = ["env_file", "format"], alias = "pyproject", hide = true)]
     pub pyproject_toml: bool,
+
+    /// Source Control Management used for this project
+    #[arg(short = 's', long = "scm", default_value = "github")]
+    pub scm: Option<String>,
```
Comment (ruben-arts) [inline src/cli/init.rs:59]: Just use the Enum directly for autocompletion, also make sure it works with lowercase and uppercase version of the strings through a test.

File: src/cli/init.rs [hunk 2]
```diff
@@ -149,9 +153,42 @@ const GITIGNORE_TEMPLATE: &str = r#"
...
+    GitHub,
+    GitLab,
+    Codeberg,
+}
+
+impl GitAttributes {
+    fn template(&self) -> &'static str {
+        match self {
+            GitAttributes::GitHub => {
+                r#"# GitHub syntax highlighting
```
Comment (ruben-arts) [inline src/cli/init.rs:166]: Lets simplify this by moving multiple options under the same result. 

```suggestion
                r#"# SCM syntax highlighting
```

**Review by ruben-arts — COMMENTED** [review]
Hi, thanks for your input!

Could you add it to the CLI documentation and add some tests? I would like to make sure it is as user friendly as possible by making it auto completing.

File: src/cli/init.rs (hunk 1 above)
Comment (alvgaona) [reply src/cli/init.rs:59]: I'll definitely do that.

File: src/cli/init.rs (hunk 2 above)
Comment (alvgaona) [reply src/cli/init.rs:166]: I'll put Codeberg and Github under the same result.

**Comment by alvgaona** [comment]
Added the use of `GitAttributes` directly in the `scm` arg. Now looking to support upper and lowercase for the input strings.

**Comment by alvgaona** [comment]
@ruben-arts would you also like to ignore case in `--format`?

**Comment by ruben-arts** [comment]
> @ruben-arts would you also like to ignore case in `--format`?

Yes please!

File: src/cli/init.rs [hunk 3]
```diff
@@ -579,4 +605,56 @@ mod tests {
...
+        ];
+
+        for (input, expected) in test_cases {
+            let args = Args::try_parse_from(&["init", "--scm", input]).unwrap();
+            assert_eq!(args.scm, Some(expected));
+        }
+    }
+
+    #[test]
+    fn test_invalid_scm_values() {
```
Comment (alvgaona) [inline src/cli/init.rs:648]: Not sure how much value this test add but it doesn't hurt to have an extra test case.

**Review by ruben-arts — APPROVED** [review]
Lovely

**Pull request merged**